packages/labs/task/src/task.ts

```diff
@@ -115,21 +115,53 @@ export class Task<
   private _onError?: (error: unknown) => unknown;
   status: TaskStatus = TaskStatus.INITIAL;
 
+  /**
+   * Controls if they task will run when its arguments change. Defaults to true.
+   */
+  autoRun = true;
+
   /**
    * A Promise that resolve when the current task run is complete.
    *
    * If a new task run is started while a previous run is pending, the Promise
    * is kept and only resolved when the new run is completed.
    */
-  taskComplete!: Promise<R>;
+  get taskComplete(): Promise<R> {
+    // True only when called for the first time or after a run is complete.
+    const lastPromiseCompletedOrFirstRequest = !this._resolveTaskComplete;
```

Comment:
I think this could be a _little_ simpler if you clear the promise state in `run()` so that if `this._taskComplete` is defined here you just always return it. Then if it's not defined you have the branches to create it with either a pending promise or a resolved one based on the status.

Reply:
you are right, MUCH cleaner! thanks!